syntax = "proto3";

package image;

// The image service definition.
service ImageService {
  // Sends an image to be stored and returns an UUID.
  rpc PostImage (PostImageRequest) returns (PostImageResponse);

  // Retrieves an image by UUID.
  rpc GetImage (GetImageRequest) returns (GetImageResponse);
}

// The request message containing the image data.
message PostImageRequest {
  bytes content = 1;
}

// The response message containing the UUID of the stored image.
message PostImageResponse {
  string uuid = 1;
}

// The request message containing the UUID.
message GetImageRequest {
  string uuid = 1;
}

// The response message containing the image file.
message GetImageResponse {
  bytes content = 1;
}
